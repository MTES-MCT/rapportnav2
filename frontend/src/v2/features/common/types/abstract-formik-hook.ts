import { FormikErrors } from 'formik'
import { ObjectSchema } from 'yup'

export interface AbstractFormikHook<T, M> {
  initValue?: M
  errors?: FormikErrors<M>
  beforeInitValue: (value?: T) => M | undefined
  beforeSubmit: (value?: M, errors?: FormikErrors<M>) => T | undefined
  handleSubmit: (
    value?: M,
    errors?: FormikErrors<M>,
    onSubmit?: (valueToSubmit?: T) => Promise<unknown>
  ) => Promise<void>
}

export interface AbstractFormikSubFormHook<M> {
  initValue?: M
  errors?: FormikErrors<M>
  validationSchema?: ObjectSchema<any>
  handleSubmit: (value?: M, errors?: FormikErrors<M>) => Promise<void>
}
