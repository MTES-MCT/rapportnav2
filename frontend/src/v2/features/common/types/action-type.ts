export enum ActionType {
  CONTROL = 'CONTROL',
  NOTE = 'NOTE',
  SURVEILLANCE = 'SURVEILLANCE',
  STATUS = 'STATUS',
  CONTACT = 'CONTACT',
  RESCUE = 'RESCUE',
  OTHER = 'OTHER',
  NAUTICAL_EVENT = 'NAUTICAL_EVENT',
  VIGIMER = 'VIGIMER',
  ANTI_POLLUTION = 'ANTI_POLLUTION',
  BAAEM_PERMANENCE = 'BAAEM_PERMANENCE',
  PUBLIC_ORDER = 'PUBLIC_ORDER',
  REPRESENTATION = 'REPRESENTATION',
  ILLEGAL_IMMIGRATION = 'ILLEGAL_IMMIGRATION',
  INQUIRY = 'INQUIRY',
  COMMUNICATION = 'COMMUNICATION',
  TRAINING = 'TRAINING',
  UNIT_MANAGEMENT_PLANNING = 'UNIT_MANAGEMENT_PLANNING',
  UNIT_MANAGEMENT_TRAINING = 'UNIT_MANAGEMENT_TRAINING',
  UNIT_MANAGEMENT_OTHER = 'UNIT_MANAGEMENT_OTHER',
  CONTROL_SECTOR = 'CONTROL_SECTOR',
  CONTROL_NAUTICAL_LEISURE = 'CONTROL_NAUTICAL_LEISURE',
  CONTROL_SLEEPING_FISHING_GEAR = 'CONTROL_SLEEPING_FISHING_GEAR',
  OTHER_CONTROL = 'OTHER_CONTROL',
  RESOURCES_MAINTENANCE = 'RESOURCES_MAINTENANCE',
  MEETING = 'MEETING',
  PV_DRAFTING = 'PV_DRAFTING',
  HEARING_CONDUCT = 'HEARING_CONDUCT',
  LAND_SURVEILLANCE = 'LAND_SURVEILLANCE',
  FISHING_SURVEILLANCE = 'FISHING_SURVEILLANCE',
  MARITIME_SURVEILLANCE = 'MARITIME_SURVEILLANCE'
}

export enum ActionGroupType {
  OTHER_GROUP = 'OTHER_GROUP',
  CONTROL_GROUP = 'CONTROL_GROUP',
  SURVEILLANCE_GROUP = 'SURVEILLANCE_GROUP',
  ADMINISTRATIVE_GROUP = 'ADMINISTRATIVE_GROUP',
  UNIT_MANAGEMENT_GROUP = 'UNIT_MANAGEMENT_GROUP'
}
