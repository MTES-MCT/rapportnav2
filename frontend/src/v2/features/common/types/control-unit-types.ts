export type ControlUnitResource = {
  id: string
  controlUnit: ControlUnit
  controlUnitId: number
  isArchived: boolean
  name: string
  note?: string
  photo?: string
  station: Station
  stationId: number
  type: ControlUnitResourceType
}

export type Station = {
  id: number
  latitude: number
  longitude: number
  name: string
}

export type ControlUnit = {
  id: number
  areaNote?: string
  administrationId: number
  departmentAreaInseeCode?: string
  isArchived: boolean
  name: string
  termsNote?: string
}

export type Administration = {
  id: number
  name: string
  controlUnitIds: number[]
  isArchived: boolean
  controlUnits: ControlUnit[]
}

enum ControlUnitResourceType {
  AIRPLANE = "AIRPLANE",
  BARGE = "BARGE",
  CAR = "CAR",
  DRONE = "DRONE",
  EQUESTRIAN = "EQUESTRIAN",
  FAST_BOAT = "FAST_BOAT",
  FRIGATE = "FRIGATE",
  HELICOPTER = "HELICOPTER",
  HYDROGRAPHIC_SHIP = "HYDROGRAPHIC_SHIP",
  KAYAK = "KAYAK",
  LIGHT_FAST_BOAT = "LIGHT_FAST_BOAT",
  MINE_DIVER = "MINE_DIVER",
  MOTORCYCLE = "MOTORCYCLE",
  NET_LIFTER = "NET_LIFTER",
  NO_RESOURCE = "NO_RESOURCE",
  OTHER = "OTHER",
  PATROL_BOAT = "PATROL_BOAT",
  PEDESTRIAN = "PEDESTRIAN",
  PIROGUE = "PIROGUE",
  RIGID_HULL = "RIGID_HULL",
  SEA_SCOOTER = "SEA_SCOOTER",
  SEMI_RIGID = "SEMI_RIGID",
  SUPPORT_SHIP = "SUPPORT_SHIP",
  TRAINING_SHIP = "TRAINING_SHIP",
  TUGBOAT = "TUGBOAT"
}
