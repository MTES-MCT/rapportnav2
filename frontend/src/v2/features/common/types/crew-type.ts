import { Service } from '@common/types/crew-types.ts'

export type Agent = {
  id: number
  firstName: string
  lastName: string
  services: Service[]
}

export type AgentRole = {
  id: string
  title: string
}

export type MissionCrewMember = {
  id?: string
  agent: Agent
  comment?: string
  role?: AgentRole
  missionId?: number
}
