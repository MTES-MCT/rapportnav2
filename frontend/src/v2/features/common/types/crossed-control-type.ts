export enum CrossControlOriginType {
  OPPORTUNITY_CONTROL = 'OPPORTUNITY_CONTROL',
  FOLLOW_UP_CONTROL = 'FOLLOW_UP_CONTROL',
  CNSP_REPORTING = 'CNSP_REPORTING',
  OTHER_REPORTING = 'OTHER_REPORTING',
  URCEM_DEDICATED_STATION = 'URCEM_DEDICATED_STATION'
}

export enum CrossControlStatusType {
  NEW = 'NEW',
  FOLLOW_UP = 'FOLLOW_UP',
  CLOSED = 'CLOSED'
}

export enum CrossControlTargetType {
  VEHICLE = 'VEHICLE',
  COMPANY = 'COMPANY'
}

export enum CrossControlConclusionType {
  NO_FOLLOW_UP = 'NO_FOLLOW_UP',
  WITH_REPORT = 'WITH_REPORT'
}

export interface CrossControl {
  id?: string
  agentId?: number
  vesselId?: number
  serviceId?: number
  nbrOfHours?: number
  endDateTimeUtc?: string
  startDateTimeUtc?: string
  isSignedByInspector?: boolean
  type?: CrossControlTargetType
  origin?: CrossControlOriginType
  status?: CrossControlStatusType
  conclusion?: CrossControlConclusionType
  vesselName?: string
  isReferentialClosed?: boolean
  vesselExternalReferenceNumber?: string
}
