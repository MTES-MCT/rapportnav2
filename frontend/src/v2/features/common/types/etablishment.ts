export interface Establishment {
  id?: string
  name?: string
  siret?: string
  address?: string
}
