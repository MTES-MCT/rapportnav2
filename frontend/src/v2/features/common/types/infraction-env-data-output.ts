import { ControlType } from '@common/types/control-types'
import { FormalNoticeEnum, InfractionTypeEnum, VesselSizeEnum, VesselTypeEnum } from '@common/types/env-mission-types'

export interface InfractionEnvTargetOutput {
  vesselIdentifier: string
  companyName: string
  relevantCourt: string
  formalNotice: FormalNoticeEnum
  toProcess: boolean
  identityControlledPerson: string
  vesselType: VesselTypeEnum
  vesselSize: VesselSizeEnum
}

export interface InfractionEnvDataOutput {
  id?: string
  missionId?: number
  actionId?: string
  controlId?: string
  controlType?: ControlType
  infractionType?: InfractionTypeEnum
  observations?: string
  natinfs?: string[]
  target?: InfractionEnvTargetOutput
}
