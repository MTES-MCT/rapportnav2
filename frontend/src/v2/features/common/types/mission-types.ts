import { ControlResource, ControlUnit } from '@common/types/control-unit-types.ts'
import { MissionCrew } from '@common/types/crew-types'
import { MissionTypeEnum, SeaFrontEnum } from '@common/types/env-mission-types'
import { Service } from '@common/types/service-types'
import { MissionAction } from './mission-action'
import { ControlUnitResource } from './control-unit-types.ts'

export enum MissionType {
  AIR = 'AIR',
  LAND = 'LAND',
  SEA = 'SEA'
}

export type MissionGeneralInfoPam = {
  id?: number
  distanceInNauticalMiles?: number
  consumedGOInLiters?: number
  consumedFuelInLiters?: number
  serviceId?: number
  nbrOfRecognizedVessel?: number
}

export type MissionULAMGeneralInfoInitial = {
  id?: number
  startDateTimeUtc: string
  endDateTimeUtc: string
  missionTypes: MissionTypeEnum[]
  missionReportType?: MissionReportTypeEnum
  reinforcementType?: MissionReinforcementTypeEnum
  nbHourAtSea?: number
}

export type MissionGeneralInfoExtended = {
  resources?: ControlResource[]
  crew?: MissionCrew[]
  isMissionArmed?: boolean
  isWithInterMinisterialService?: boolean
  observations?: string
  isAllAgentsParticipating?: boolean
  interMinisterialServices?: InterMinisterialService[]
  isUnderJdp?: boolean
  jdpType?: JdpTypeEnum
}

export enum MissionReportTypeEnum {
  FIELD_REPORT = 'FIELD_REPORT',
  OFFICE_REPORT = 'OFFICE_REPORT',
  EXTERNAL_REINFORCEMENT_TIME_REPORT = 'EXTERNAL_REINFORCEMENT_TIME_REPORT'
}

export enum JdpTypeEnum {
  DOCKED = 'DOCKED',
  ONBOARD = 'ONBOARD',
}

export enum MissionReinforcementTypeEnum {
  PATROL = 'PATROL',
  OTHER_ULAM = 'OTHER_ULAM',
  SEA_TRAINER = 'SEA_TRAINER',
  OTHER = 'OTHER',
  DIRM = 'DIRM'
}

export enum MissionSourceEnum {
  MONITORENV = 'MONITORENV',
  MONITORFISH = 'MONITORFISH',
  POSEIDON_CACEM = 'POSEIDON_CACEM',
  POSEIDON_CNSP = 'POSEIDON_CNSP',
  RAPPORTNAV = 'RAPPORTNAV'
}

export enum CompletenessForStatsStatusEnum {
  COMPLETE = 'COMPLETE',
  INCOMPLETE = 'INCOMPLETE'
}

export enum MissionStatusEnum {
  UPCOMING = 'UPCOMING',
  IN_PROGRESS = 'IN_PROGRESS',
  ENDED = 'ENDED',
  UNAVAILABLE = 'UNAVAILABLE'
}

export type Mission = {
  id?: number
  missionTypes: MissionType[]
  controlUnits: ControlUnit[]
  openBy?: string
  completedBy?: string
  observationsCacem?: string
  observationsCnsp?: string
  facades?: string
  geom?: string
  startDateTimeUtc: string
  endDateTimeUtc?: string
  missionSource: MissionSourceEnum
  hasMissionOrder: boolean
  isUnderJdp: boolean
  isGeometryComputedFromControls: boolean
}

export type CompletenessForStats = {
  sources?: MissionSourceEnum[]
  status?: CompletenessForStatsStatusEnum
}

export type MissionEnvData = {
  controlUnits: Omit<ControlUnit, 'id'>[]
  endDateTimeUtc?: string
  facade: SeaFrontEnum
  geom?: Record<string, any>[]
  hasMissionOrder?: boolean
  isUnderJdp?: boolean
  missionSource: MissionSourceEnum
  missionTypes: MissionTypeEnum[]
  observationsCacem?: string
  observationsCnsp?: string
  observationsByUnit?: string
  openBy: string
  startDateTimeUtc: string
}

export type MissionGeneralInfo2 = MissionGeneralInfoPam & {
  missionReportType?: MissionReportTypeEnum
  missionTypes: MissionTypeEnum[]
  reinforcementType?: MissionReinforcementTypeEnum
  nbHourAtSea?: number
  crew?: MissionCrew[]
  services?: Service[]
  isMissionArmed?: boolean
  isWithInterMinisterialService?: boolean
  isAllAgentsParticipating?: boolean
  missionId?: number
  startDateTimeUtc: string
  endDateTimeUtc: string
  observations?: string
  resources?: ControlResource[]
  interMinisterialServices?: InterMinisterialService[]
  missionIdString?: string
  isUnderJdp?: boolean
  jdpType?: JdpTypeEnum
}

export type Mission2 = {
  id: number
  isCompleteForStats: boolean
  status: MissionStatusEnum
  envData: MissionEnvData
  generalInfos: MissionGeneralInfo2
  completenessForStats?: CompletenessForStats
  actions: MissionAction[]
  missionIdString?: string
}

export type MissionListItem = {
  id: number
  openBy?: string
  exportLabel?: string
  status?: MissionStatusEnum
  missionNameUlam?: string
  startDateTimeUtc: string
  startDateTimeUtcText?: string
  crew?: MissionCrew[]
  crewNumber?: string
  endDateTimeUtc?: string
  endDateTimeUtcText?: string
  missionNamePam?: string
  observationsByUnit?: string
  missionSource?: MissionSourceEnum
  completenessForStats?: CompletenessForStats
  controlUnits?: ControlUnit[]
  resources?: ControlUnitResource[],
  missionReportType?: MissionReportTypeEnum
  missionIdString?: string
  isUnderJdp?: boolean
  jdpType?: JdpTypeEnum
}

export type InterMinisterialService = {
  administrationId: number
  controlUnitId: number
}
