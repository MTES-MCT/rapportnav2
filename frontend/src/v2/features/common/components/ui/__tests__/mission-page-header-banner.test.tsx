import { MissionSourceEnum } from '@common/types/env-mission-types'
import { render } from '../../../../../../test-utils'
import { CompletenessForStatsStatusEnum } from '../../../types/mission-types'
import MissionPageHeaderBanner from '../mission-page-header-banner'

describe('MissionPageHeaderBanner', () => {
  it('should match the snapshot', () => {
    const completenessForStats = {
      sources: [MissionSourceEnum.MONITORENV],
      status: CompletenessForStatsStatusEnum.INCOMPLETE
    }
    const wrapper = render(<MissionPageHeaderBanner completenessForStats={completenessForStats} />)
    expect(wrapper).toMatchSnapshot()
  })
})
