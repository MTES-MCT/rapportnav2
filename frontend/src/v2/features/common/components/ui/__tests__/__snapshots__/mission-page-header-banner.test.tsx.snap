// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MissionPageHeaderBanner > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-cYoYeT foUlcF Component-Banner"
      >
        <div
          class="sc-ktvNQe bxHGws"
        >
          <p>
            La mission est terminée mais le rapport n'est pas complet : des données sont manquantes dans des champs à remplir par le CACEM.
          </p>
        </div>
        <div
          class="sc-dslWvo dzrzey banner-button"
          title="masquer"
        >
          <button
            class="sc-lkbiLU jLIjij Element-LinkButton sc-eXfTSZ clGZzi"
          >
            Masquer
          </button>
        </div>
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-cYoYeT foUlcF Component-Banner"
    >
      <div
        class="sc-ktvNQe bxHGws"
      >
        <p>
          La mission est terminée mais le rapport n'est pas complet : des données sont manquantes dans des champs à remplir par le CACEM.
        </p>
      </div>
      <div
        class="sc-dslWvo dzrzey banner-button"
        title="masquer"
      >
        <button
          class="sc-lkbiLU jLIjij Element-LinkButton sc-eXfTSZ clGZzi"
        >
          Masquer
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
